src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -235,36 +245,27 @@ impl Service<PartitionInnerBuffer<Vec<InputLogEvent>, CloudwatchKey>>
         let svc = if let Some(svc) = &mut self.clients.get_mut(&key) {
             svc.clone()
         } else {
-            let svc = {
-                let policy = self.request_settings.retry_policy(CloudwatchRetryLogic);
-
-                let cloudwatch = CloudwatchLogsSvc::new(&self.config, &key, self.client.clone());
-                let timeout = Timeout::new(cloudwatch, self.request_settings.timeout);
-
-                let buffer = Buffer::new(timeout, 1);
-                let retry = Retry::new(policy, buffer);
-
-                let rate = Rate::new(
+            let svc = ServiceBuilder::new()
+                .buffer(self.request_settings.in_flight_limit)
```

Comment:
Why we had buffer size 1 here? It looks incorrect and on update `cloudwatch_insert_log_event_batched` test showed this.

Reply:
Good question. It's a little confusing here whether those limits should apply per key (i.e. stream) or for the sink as a whole, from the user's perspective. On the implemenation side, there is also the complication that requests within each stream require a token from the previous request, which requires a concurrency limit of one.

Reply:
This concurrency limit should never be hit also since the buffer is of size 1 and @lukesteensen point about the stream being serial is why we originally wrote it this way.